### DISCUSSION TITLE
Victron Wechselrichter und smartshut anschließen

### DISCUSSION THREAD

**Discussion by rainer667** [hoylabs/OpenDTU-OnBattery#972] [Q&A]
Hallo Leute,
erstmal danke für die tolle Software und Weiterentwicklung.
Ich habe mich von den Victron Geräten überzeugen lassen und hab mir einen Smartsolar Mppt und den Smartshut gekauft, mit Bluetooth gekoppelt und angeschlossen. Das funktioniert soweit echt gut und kann gut die Daten mit Bluetooth lesen.
Doch nun bin ich alter Mann mit meinem Talent und schlechtem Englisch am Ende wenn's um das verbinden mit der DTU on Battery geht. 
Lt. der Doku (bin mir leider nicht sicher) hab ich am ESP32 den Pin 21 und 22, 3.3V mit Grd was an den  Smartsolar Mppt über den Pegelwandler ADUM1201 anzuschließen ist. Wie wird der SmartShunt angeschlossen? Und muss ich noch eine Datei mit der Pinbelegung machen, einspielen?
Ich hoffe ihr bringt mich auf Kurs. Evtl. könnte das im Wiki ergänzt werden. Ich könnte mir vorstellen dass damit einige Fragen entfallen.
Vielen Dank Rainer

**Comment by SW-Niko** [comment]
Hallo @rainer667,
such mal im Wiki unter:  https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/VE.Direct-Interface
dort sollte alles stehen was du benötigst um die Geräte mit OpenDTU-OnBattery zu verbinden.

**Comment by Manos1966** [comment]
@rainer667 
Hmmm... ich haette schwoeren koennen, wir haben SmartSHUNT in das Wiki beschrieben, ich finde es aber nicht 🤦‍♂️ 

Also: Ein primitives Bild der moeglichen Anschluessen ist gut versteckt unter: https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Huawei-AC-PSU
Dieses Bild hat ein paar "Fehler", naemlich die VE.Direkt Verbindung zwischen Victron und OpenDTU: 

- "Kenner" wissen, dass eine Verbindung wird zwischen "Sender" und "Empfaenger" (Tx zu Rx / Rx zu Tx)  "Crossover"
- "Normale Sterbliche" (wie ich) denken Digital und wuerden Tx mit Tx und Rx zu Rx verbinden "Direkt"

Beide Versionen funktionieren, abhaengig wie das Verbindungs-Kabel gemacht wird: Direkt (Tx zu Tx/Rx zu Rx) oder Crossover (Tx zu Rx/Rx zu Tx) hier erklaert https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/VE.Direct-Interface#cables

Wie auch immer, unten Links siehst du SmartSHUNT und die PIN Belegung (als Batterie) 
![Wiring_ESP32_Symbol - Victr1Victr2PylontechHuaweiRelaisCMT2300updRJ45](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/24e0969d-dfd7-4b21-bffd-b262d49ebc9d)

SmartSHUNT ist eine Auswahl unter Batterie:
![SmartSHUNT](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/ead9e625-fa00-4594-bc65-9119dcfda67c)

**Comment by schlimmchen** [comment]
> Beide Versionen funktionieren, abhaengig wie das Verbindungs-Kabel gemacht wird

Naja, nicht wirklich. Es muss schon der TX-Pin des Victron als RX Pin am ESP32 definiert werden, und optional der RX-Pin des Victron als TX-Pin am ESP32.

> Pegelwandler ADUM1201

Sehr gut, dass du einen ADUM1201 verwendest.

SmartShunt und Victron MPPT werden jeweils als VE.Direct angeschlossen. Du musst mindestens den jeweiligen TX-Pin des Gerätes an einen Pin des ESP32 verbinden.

> Und muss ich noch eine Datei mit der Pinbelegung machen, einspielen?

Ja, siehe [Device Profiles im Wiki](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Device-Profiles)

Hast du noch eine konkrete Frage?

**Comment by rainer667** [comment]
@SW-Niko  Danke, nur das hab ich leider nicht richtig verstanden und deshalb gefragt.
@Manos1966  Danke, wie immer schnelle und gute Antwort, nur weiß ich immer noch nicht was ich machen soll.
@schlimmchen Danke für die tolle Software, die Arbeit und Zeit die du investierst.
Ich bin ehrlich gesagt am verzweifeln. 
Ich weiß nicht welche Pins am esp32 für den Wechselrichter und welche Pins für den Shunt sind. Ich hab schon versucht Fusionplatine mit der neuen Zusatzplatine zu erwerben, nur hüllt sich gute Mann auch in Schweigen.
Opensource ist Klasse, wenn man sie verwenden kann. Wenn ich dazu ein Studium brauche werden nicht viele Gefallen dran finden.  
Ich denke das viele wie ich einen Victron Wechselrichter und den Shunt haben. Für die wäre es eine Erleichterung wenn jemand das Wiki bitte ergänzen würde welche Pins verwendet werden sollen? Auch eine kleine Skizze dazu wie der ADUM1201 mit den Pins anzuschliessen ist. Auch weiß ich nicht ob ich einem ADUM1201 oder zwei brauche. Wahrscheinlich alles Fragen um die euch keiner einen Gedanken verliert, nur gibt's auch Dummies wie mich.
VG Rainer

**Comment by schlimmchen** [comment]
> Ich weiß nicht welche Pins am esp32 für den Wechselrichter und welche Pins für den Shunt sind.

Das darfst du dir aussuchen, du musst es dann aber im DeviceProfile festlegen, welche du ausgesucht hast. Wenn du eine OpenDTU FUSION Platine hast, dann sind die beiden UART RX/TX Pins links oben an der Platine zu empfehlen. Welchen GPIO Nummer die haben, steht in der [Doku der OpenDTU FUSION](https://github.com/markusdd/OpenDTUFusionDocs) Platine.

> Ich hab schon versucht Fusionplatine mit der neuen Zusatzplatine zu erwerben, nur hüllt sich gute Mann auch in Schweigen.

Die Zusatzplatine ist ja auch noch nicht verfügbar und Unterstützung leistet AllicanceApps ausdrücklich nicht. Das musst du akzeptieren.

> Wenn ich dazu ein Studium brauche werden nicht viele Gefallen dran finden.

Es verwenden Hunderte Leute OpenDTU-OnBattery, und die allerwenigstens brauchen überhaupt Unterstützung. Es liegt nicht am fehlenden Studium. Wenn man ein Hobby-Projekt wie die Inbetriebnahme eines OpenDTU-OnBattery angeht, dann muss man auch ein paar Grundkenntnisse mitbringen oder das passende Durchhaltevermögen und das Geschick, weiterführende Infos im Internet zu finden.

> Auch eine kleine Skizze dazu wie der ADUM1201 mit den Pins anzuschliessen ist.

Das steht im Wiki, Niko hat den passenden Link bereits geteilt. Nutze einen Online-Übersetzer, wenn dir das Englisch Schwierigkeiten bereitet.

> Auch weiß ich nicht ob ich einem ADUM1201 oder zwei brauche.

Einen pro VE.Direkt-Gerät.

> Danke für die tolle Software, die Arbeit und Zeit die du investierst.

Bitte :heart:

**Reply by rainer667** [reply]
Danke, das ist opensource

**Comment by Manos1966** [comment]
Hallo @rainer667 
Bist du weiter gekommen, oder immer noch nicht?

**Comment by Manos1966** [comment]
Hast du PIN 27 und 14 Frei? Dann siehe im Bild
- wohin 27 und 14 angeschlossen werden
- was du in der PIN Konfiguration der OpenDTU eingeben muss (schwarzes Bild) 

![SmartSHUN PIN MappingT](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/c0da65d6-002c-4ee6-ab28-e22af8947919)

**Comment by Matti07** [comment]
> > Ich hab schon versucht Fusionplatine mit der neuen Zusatzplatine zu erwerben, nur hüllt sich gute Mann auch in Schweigen.
> 
> Die Zusatzplatine ist ja auch noch nicht verfügbar und Unterstützung leistet AllicanceApps ausdrücklich nicht. Das musst du akzeptieren.

Die Zusatzplatine ist lt. dem Support von allianceapps in ca. 3 Wochen verfügbar.
